Unable to compile on windows using Go 1.15

- Gitea version (or commit ref): 1.12.3+16-g4a548a033 AND 1.13.0+dev-464-g63f3c42be
- Git version: 2.28.0.windows.1
- Operating system: Windows 10 2004 Pro x64
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [x] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

When rebuilding Gitea v1.12 for Go v1.15.0, I get a make error.

```
...
bindata for options already up-to-date
bindata for public already up-to-date
bindata for templates already up-to-date
CGO_CFLAGS="-g -O2 -DSQLITE_MAX_VARIABLE_NUMBER=32766" go build -mod=vendor -v  -tags 'bindata sqlite sqlite_unlock_notify' -ldflags '-s -w  -X "main.MakeVersion=GNU Make 4.3" -X "main.Version=1.12.3+16-g4a548a033" -X "main.Tags=bindata sqlite sqlite_unlock_notify"' -o gitea.exe
...
code.gitea.io/gitea/routers/admin
code.gitea.io/gitea/routers/api/v1/repo
code.gitea.io/gitea/routers/api/v1/user
code.gitea.io/gitea/routers/api/v1/admin
code.gitea.io/gitea/routers/api/v1/org
code.gitea.io/gitea/routers/api/v1
code.gitea.io/gitea/routers/routes
code.gitea.io/gitea/cmd
code.gitea.io/gitea
# code.gitea.io/gitea
C:\Tools\Go\pkg\tool\windows_amd64\link.exe: running gcc failed: exit status 1
C:/Tools/mingw-w64/x86_64-8.1.0-posix-seh-rt_v6-rev0/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/8.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: Error: export ordinal too large: 79462
collect2.exe: error: ld returned 1 exit status

make: *** [Makefile:521: gitea.exe] Error 2
```

When rebuilding Gitea v1.13 for Go v1.15.0, I get a make error as well:

```
...
generating bindata for options
writing bindata.go
generating bindata for public
writing bindata.go
generating bindata for templates
writing bindata.go
CGO_CFLAGS="-g -O2 -DSQLITE_MAX_VARIABLE_NUMBER=32766" go build -mod=vendor -v  -tags 'bindata sqlite sqlite_unlock_notify' -ldflags '-s -w  -X "main.MakeVersion=GNU Make 4.3" -X "main.Version=1.13.0+dev-464-g63f3c42be" -X "main.Tags=bindata sqlite sqlite_unlock_notify"' -o gitea.exe
...
code.gitea.io/gitea/routers
code.gitea.io/gitea/routers/admin
code.gitea.io/gitea/routers/api/v1/repo
code.gitea.io/gitea/routers/api/v1/user
code.gitea.io/gitea/routers/api/v1/admin
code.gitea.io/gitea/routers/api/v1/org
code.gitea.io/gitea/routers/api/v1
code.gitea.io/gitea/routers/routes
code.gitea.io/gitea/cmd
code.gitea.io/gitea
# code.gitea.io/gitea
C:\Tools\Go\pkg\tool\windows_amd64\link.exe: running gcc failed: exit status 1
C:/Tools/mingw-w64/x86_64-8.1.0-posix-seh-rt_v6-rev0/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/8.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: Error: export ordinal too large: 96057
collect2.exe: error: ld returned 1 exit status

make: *** [Makefile:545: gitea.exe] Error 2
```

## Screenshots

N/A

--- a/.drone.yml
+++ b/.drone.yml
@@ -548,8 +548,7 @@ steps:
     pull: always
     image: techknowlogick/xgo:go-1.15.x
     commands:
-      - apt update && apt -y install curl
-      - curl -sL https://deb.nodesource.com/setup_12.x | bash - && apt -y install nodejs
+      - curl -sL https://deb.nodesource.com/setup_14.x | bash - && apt -y install nodejs
       - export PATH=$PATH:$GOPATH/bin
       - make release
     environment:

--- a/.drone.yml
+++ b/.drone.yml
@@ -447,23 +447,13 @@ steps:
     commands:
       - git fetch --tags --force
 
-  - name: static-windows
-    pull: always
-    image: techknowlogick/xgo:go-1.14.x
-    commands:
-      - apt update && apt -y install curl
-      - curl -sL https://deb.nodesource.com/setup_12.x | bash - && apt -y install nodejs
-      - export PATH=$PATH:$GOPATH/bin
-      - make frontend generate release-windows
-    environment:
-      GOPROXY: https://goproxy.cn # proxy.golang.org is blocked in China, this proxy is not
-      TAGS: bindata sqlite sqlite_unlock_notify
-
   - name: static
     pull: always
     image: techknowlogick/xgo:go-1.15.x
     commands:
-      - make release-linux release-darwin release-copy release-compress release-sources release-docs release-check
+      - curl -sL https://deb.nodesource.com/setup_14.x | bash - && apt -y install nodejs
+      - export PATH=$PATH:$GOPATH/bin
+      - make release
     environment:
       GOPROXY: https://goproxy.cn # proxy.golang.org is blocked in China, this proxy is not
       TAGS: bindata sqlite sqlite_unlock_notify

--- a/Makefile
+++ b/Makefile
@@ -42,8 +42,10 @@ ifeq ($(HAS_GO), GO)
 endif
 
 ifeq ($(OS), Windows_NT)
+	GOFLAGS := -v -buildmode=exe
 	EXECUTABLE ?= gitea.exe
 else
+	GOFLAGS := -v
 	EXECUTABLE ?= gitea
 endif
 
@@ -55,7 +57,6 @@ endif
 
 GOFMT ?= gofmt -s
 
-GOFLAGS := -v
 EXTRA_GOFLAGS ?=
 
 MAKE_VERSION := $(shell $(MAKE) -v | head -n 1)
@@ -556,7 +557,7 @@ release-windows: | $(DIST_DIRS)
 		GO111MODULE=off $(GO) get -u src.techknowlogick.com/xgo; \
 	fi
 	@echo "Warning: windows version is built using golang 1.14"
-	CGO_CFLAGS="$(CGO_CFLAGS)" GO111MODULE=off xgo -go go-1.14.x -dest $(DIST)/binaries -tags 'netgo osusergo $(TAGS)' -ldflags '-linkmode external -extldflags "-static" $(LDFLAGS)' -targets 'windows/*' -out gitea-$(VERSION) .
+	CGO_CFLAGS="$(CGO_CFLAGS)" GO111MODULE=off xgo -go $(XGO_VERSION) -buildmode exe -dest $(DIST)/binaries -tags 'netgo osusergo $(TAGS)' -ldflags '-linkmode external -extldflags "-static" $(LDFLAGS)' -targets 'windows/*' -out gitea-$(VERSION) .
 ifeq ($(CI),drone)
 	cp /build/* $(DIST)/binaries
 endif

